client/router_update.go

```diff
@@ -515,6 +529,19 @@ func (cli *VanClient) RouterUpdateVersionInNamespace(ctx context.Context, hup bo
 			}
 		}
 	}
+
+	if updateRouterPolicyRule {
+		err = kube.UpdateRole(namespace, types.TransportRoleName, types.TransportPolicyRule, cli.KubeClient)
+		if err != nil {
+			return false, err
+		}
+	}
+
+	if addCertsSharedVolume {
+		kube.AppendSharedVolume(&router.Spec.Template.Spec.Volumes, &router.Spec.Template.Spec.Containers[0].VolumeMounts, &router.Spec.Template.Spec.Containers[1].VolumeMounts, "skupper-router-certs", "/etc/skupper-router-certs")
+		updateRouter = true
+	}
```

Comment:
This must be done before line 435: `if updateRouter || updateSite || hup`
Otherwise this change is not applied to the skupper-router deployment.

Reply:
Thanks! I will fix it.